From 52545628c07be2fd1c9df598a17130d92f12da23 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 20 Nov 2023 15:17:34 +0000
Subject: [PATCH] ASoC: dwc: Remove check in set_bclk_ratio handling

A check added to dw_i2s_set_bclk_ratio that the data format is
consistent with the ratio seems reasonable but breaks when the
ratio is changed before the format. Remove the check - it is
unnecessary.

See: https://github.com/raspberrypi/linux/issues/5724
Fixes: 9c6694c24f26 ("ASOC: dwc: Fix 16-bit audio handling")

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 sound/soc/dwc/dwc-i2s.c | 3 ---
 1 file changed, 3 deletions(-)

--- a/sound/soc/dwc/dwc-i2s.c
+++ b/sound/soc/dwc/dwc-i2s.c
@@ -427,11 +427,8 @@ static int dw_i2s_set_bclk_ratio(struct
 				 unsigned int ratio)
 {
 	struct dw_i2s_dev *dev = snd_soc_dai_get_drvdata(cpu_dai);
-	struct i2s_clk_config_data *config = &dev->config;
 
 	dev_dbg(dev->dev, "%s(%d)\n", __func__, ratio);
-	if (ratio < config->data_width * 2)
-		return -EINVAL;
 
 	switch (ratio) {
 	case 32:
